check-pricing.report.md

```diff
@@ -0,0 +1,226 @@
+# Pricing Report
+
+**Date: March 2025**
+
+## Motivation
+
+The cost of operating a canister has increased and some prices no longer cover the cost of a typical API call:
+
+```
+OK: Signer balance rose by 945_027_353 for: schnorr_public_key
+OK: Signer balance rose by 13_791_062_383 for: schnorr_sign
+WARNING: signer balance fell by -35_728_568 for: btc_caller_address
+WARNING: signer balance fell by -40_248_266 for: btc_caller_balance
+OK: Signer balance rose by 35_997_576_554 for: btc_caller_send
+OK: Signer balance rose by 945_500_972 for: eth_address
+OK: Signer balance rose by 13_745_862_039 for: eth_personal_sign
+OK: Signer balance rose by 13_715_928_103 for: eth_sign_prehash
+OK: Signer balance rose by 13_713_932_473 for: eth_sign_transaction
+OK: Signer balance rose by 945_149_949 for: generic_caller_ecdsa_public_key
+OK: Signer balance rose by 13_791_161_094 for: generic_sign_with_ecdsa
+```
+
+## Current pricing
+
+The cost in cycles for each method is:
+
+```
+            SignerMethods::GenericCallerEcdsaPublicKey
+            | SignerMethods::SchnorrPublicKey
+            | SignerMethods::EthAddress
+            | SignerMethods::EthAddressOfCaller => 1_000_000_000,
+            SignerMethods::GenericSignWithEcdsa
+            | SignerMethods::SchnorrSign
+            | SignerMethods::EthSignTransaction
+            | SignerMethods::EthPersonalSign
+            | SignerMethods::EthSignPrehash => 40_000_000_000,
+            SignerMethods::BtcCallerAddress => 20_000_000,
+            SignerMethods::BtcCallerBalance => 40_000_000,
+            SignerMethods::BtcCallerSend => 130_000_000_000,
```

Comment:
Note: These were added to the JSON below manually, which is not ideal.  It would be useful to make the pricing data available via an API.  Then a simple `dfx canister call signer pricelist | idl2json > pricing.json` would give reliable numbers.